rcjaRegistration/events/views.py

```diff
@@ -234,3 +236,103 @@ def delete(self, request, teamID=None, attendeeID=None, eventID=None):
+                teamNumber += 1
+                students = Student.objects.filter(team=attendance.childObject())
+                for student in students:
+                    studentNumber += 1
+                    if student.gender == "male":
+                        maleNumber += 1
+                    elif student.gender == "female":
+                        femaleNumber += 1
+                    else:
+                        otherNumber += 1
+            elif attendance.eventAttendanceType() == "workshopattendee": # Workshop
+                if attendance.attendeeType == "student":
+                    studentNumber += 1
+                else:
+                    teacherNumber += 1
+                if attendance.gender == "male":
+                    maleNumber += 1
+                elif attendance.gender == "female":
+                    femaleNumber += 1
+                else:
+                    otherNumber += 1
+
+        # Create output
+        if event.eventType == "competition":
+            if studentNumber > 0:
+                mPercent = round(maleNumber/studentNumber*100)
+                fPercent = round(femaleNumber/studentNumber*100)
+                oPercent = round(otherNumber/studentNumber*100)
+            else:
+                mPercent, fPercent, oPercent = [0,0,0]
+            eventDict["participants_one"] = f"Teams: {teamNumber}"
+            eventDict["participants_two"] = f"Students: {studentNumber}"
+            eventDict["participants_three"] = f"{fPercent}%F, {mPercent}%M, {oPercent}% other"
+        else: # Workshop
+            if studentNumber + teacherNumber > 0:
+                mPercent = round(maleNumber/(studentNumber+teacherNumber)*100)
+                fPercent = round(femaleNumber/(studentNumber+teacherNumber)*100)
+                oPercent = round(otherNumber/(studentNumber+teacherNumber)*100)
+            else:
+                mPercent, fPercent, oPercent = [0,0,0]
+            eventDict["participants_one"] = f"Students: {studentNumber}"
+            eventDict["participants_two"] = f"Teachers: {teacherNumber}"
+            eventDict["participants_three"] = f"{fPercent}%F, {mPercent}%M, {oPercent}% other"
+
+        if event.venue != None:
+            eventDict["location"] = event.venue.name
+        else:
+            eventDict["location"] = "None"
+        
+        events.append(eventDict)
+
+    return events
+
+@login_required
+def summaryReport(request):
+    user_state = request.user.currentlySelectedAdminState
+    year = request.user.currentlySelectedAdminYear
+    if request.method == 'GET':
+        form = getSummaryForm(request)
+        if form.is_valid():
+            # Save user
+            user_state = form.cleaned_data["state"]
+            year = form.cleaned_data["year"]
+        events = getEventsForSummary(user_state, year)
+
+    context = {
+        "events": events,
+        "form": form,
+        'state': State.objects.filter(id = user_state)[0].name,
```

Comment:
Hey Harry - I am attempting to get this running on my own machine, but it throws an error here

![image](https://github.com/user-attachments/assets/8496b0d9-31c5-47d2-866f-d3b6ebf1fa22)

I have my my admin state/year unset (set to "All"), if I set them then it can continue onwards. 
![image](https://github.com/user-attachments/assets/fe452eb1-53d3-4b6a-8a4d-8b0cfd17cd69)

Perhaps the page needs to have a step that doesn't render the table and only shows the state/year selectors if they aren't already set?

Reply:
Another related error
![image](https://github.com/user-attachments/assets/4b9590b7-374d-4ada-813b-93282992b277)